src/provider.ts

```diff
@@ -71,18 +71,23 @@ export function resolveProvider (nuxt: any, key: string, input: InputProvider):
   }
 }
 
+const providerFlags = {
+  vercel: [process.env.VERCEL, process.env.NUXT_ENV_VERCEL_ENV, process.env.NOW_BUILDER].some(Boolean)
+}
+
 export function detectProvider (userInput?: string) {
   if (process.env.NUXT_IMAGE_PROVIDER) {
     return process.env.NUXT_IMAGE_PROVIDER
   }
 
   if (userInput && userInput !== 'auto') {
+    for (const [provider, enabled] of Object.entries(providerFlags)) {
+      if (userInput === provider && !enabled) {
+        return 'static'
+      }
+    }
     return userInput
   }
 
-  if (process.env.VERCEL) {
```

Comment:
This regression needs to be fixed regardless and I think simply checking an inline env was enough. Users don't need to locally configure provider to `vercel` since it is supposed to auto-detected (thus default is `static` for local) We can introduce something like `devProvider` option to allow possibility of overriding for dev time and explicitly use a provider for prod

Reply:
@pi0 I've made a stab at a development mode for vercel but it sounds like you have something slightly different in mind - happy to revert and we can discuss better approach in separate PR if you think better.

Reply:
This good track to me